agents-api/agents_api/routers/responses/get_response.py

```diff
@@ -0,0 +1,143 @@
+from typing import Annotated
+from uuid import UUID
+
+from fastapi import Depends, HTTPException, Query
+
+from ...autogen.Entries import History
+from ...autogen.openapi_model import (
+    Includable,
+    InputTokensDetails,
+    OutputMessage,
+    OutputText,
+    OutputTokensDetails,
+    ResponseUsage,
+    Text,
+)
+from ...autogen.Responses import (
+    ComputerToolCall,
+    FileSearchToolCall,
+    FunctionToolCall,
+    ReasoningItem,
+    Response,
+    ResponseFormatText,
+    WebSearchToolCall,
+)
+from ...common.protocol.developers import Developer
+from ...dependencies.developer_id import get_developer_data
+from ...queries.entries.get_history import get_history as get_history_query
+from .router import router
+
+
+@router.get("/responses/{response_id}", tags=["responses"])
+async def get_response(
+    response_id: UUID,
+    developer: Annotated[Developer, Depends(get_developer_data)],
+    include: Annotated[list[Includable], Query()] = [],
+) -> Response:
+    # TODO: Continue the implementation of the logic to get a response by id
+
+    session_id = response_id
+
+    session_history: History = await get_history_query(
+        developer_id=developer.id, session_id=session_id
+    )
+
+    if not session_history.entries:
+        raise HTTPException(status_code=404, detail="Provided response id does not exist")
+
+    last_entries = []
+
+    for i, entry in enumerate(reversed(session_history.entries)):
+        if entry.role in ["user", "developer"]:
+            # Take all entries after the current entry
+            last_entries = session_history.entries[len(session_history.entries) - i :]
```

Comment:
we can use negative indexes here

Reply:
![image](https://github.com/user-attachments/assets/a1f2fcc8-4fb6-45e0-bc4c-01eb3052ee53)